src/GraphCoarsenBuilder.cpp

```diff
@@ -69,6 +93,10 @@ void AssembleMBuilder::RegisterRow(int agg_index, int row, int cdof_loc, int bub
     bubble_counter_ = bubble_counter;
 }
 
+void CoefficientMBuilder::RegisterRow(int agg_index, int row, int cdof_loc, int bubble_counter)
+{
+}
```

Comment:
Instead of all of these empty implementations, the base class `CoarseMBuilder` can just have the empty implementation (instead of being pure virtual).  Then derived classes can override the implementation if they need something specific done.  Maybe keeping `Setup` as the only pure virtual function.  Should simplify the current design a bit.

Reply:
Thanks, done.